homeassistant/components/fritzbox/climate.py

```diff
@@ -93,9 +93,10 @@ def current_temperature(self):
     @property
     def target_temperature(self):
         """Return the temperature we try to reach."""
-        if self._target_temperature in (ON_API_TEMPERATURE,
-                                        OFF_API_TEMPERATURE):
-            return None
+        if self._target_temperature == ON_API_TEMPERATURE:
```

Comment:
I don't know about this. It was already like this before: https://github.com/home-assistant/home-assistant/blob/0.95.0/homeassistant/components/fritzbox/climate.py#L91-L97

Also, this would mean target temp is hardcoded to 30 C?

Reply:
Yes, I did not change that code

Reply:
The way I see it is that 127.0 and 126.5 are magic numbers where 'on' would equal 'boost' and 'off'... well ... 'off'. ;-)
So either it's one of the magic numbers, a temp equal either eco or comfort (which would translate to 'auto') or something else which is 'manual'.

This is from the upstream module:
```python
            return {
                126.5: 'off',
                127.0: 'on',
                self.eco_temperature: 'eco',
                self.comfort_temperature: 'comfort'
            }[self.target_temperature]
        except KeyError:
            return 'manual'
```

Reply:
But we should not report the magic numbers in our frontend. They are internal and it's not actually the temperature that we want to reach. Imagine looking at your thermostat and it said 125 ;)

Reply:
It does not, it translates to 30.0/0.0.

Reply:
But does that make sense ?

Reply:
Why not? 0.0/30.0 are the values that the upstream module defines as valid range for setting temperature. Those correspond to off/boost in preset terms. To me it makes perfect sense to use that in HA to show what the thermostat is set to and further translate that to the HVAC_MODE, being either off or heat.

See: https://github.com/hthiery/python-fritzhome/blob/b9d44918f015b491ed631cbf6e63884f6a25cea2/pyfritzhome/fritzhome.py#L202-L210